hypershift-operator/controllers/hostedcluster/internal/platform/kubevirt/kubevirt.go

```diff
@@ -154,7 +165,30 @@ func (p Kubevirt) CAPIProviderDeploymentSpec(hcluster *hyperv1.HostedCluster, _
 func (p Kubevirt) ReconcileCredentials(ctx context.Context, c client.Client, createOrUpdate upsert.CreateOrUpdateFN,
 	hcluster *hyperv1.HostedCluster,
 	controlPlaneNamespace string) error {
-	return nil
+
+	// If external infra cluster kubeconfig has been provided, copy the secret from the "clusters" to the hosted control plane namespace
+	kvPlatform := hcluster.Spec.Platform.Kubevirt
+	if kvPlatform == nil || kvPlatform.ExtClusterKubeConfigSecret == nil {
+		return nil
+	}
+
+	var source corev1.Secret
+	name := client.ObjectKey{Namespace: hcluster.Namespace, Name: hcluster.Name + extClusterCredentialsSuffix}
+	if err := c.Get(ctx, name, &source); err != nil {
+		return fmt.Errorf("failed to get secret %s: %w", name, err)
+	}
+
+	target := &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Namespace: controlPlaneNamespace, Name: name.Name}}
+	_, err := createOrUpdate(ctx, c, target, func() error {
+		if target.Data == nil {
+			target.Data = map[string][]byte{}
+		}
+		for k, v := range source.Data {
+			target.Data[k] = v
+		}
+		return nil
+	})
+	return err
```

Comment:
The second part of naming the credentials secret in the HCP namespace `kubevirt-infra-credentials` is reasonable, since there is only one secret of that kind in an HCP namespace.
But the credentials secret in the hosted cluster namespace (`clusters`) should be unique and contain the hcp name in its name.

I'm not quite sure how manually creating a credentials secret by the user at the `clusters` namespace with an arbitrary name (e.g. `my-super-secret`) is going to be caught and copied to the relevant HCP namespace by hypershift-operator.

Reply:
> ok, now i understood you meant the users will interact directly with the HostedCluster API and not using the hypershift CLI.
> IMHO, it makes more sense that the secrets names are aligned between clusters ns and the HCP (clusters-<guest_cluster_name>) namespace, that there will be a 1:1 relation between them.

I'm slightly worried about naming collisions in the HCP namespace. For example, say someone named their kubevirt infra secret `admin-kubeconfig` in the HC namespace. If we copied `admin-kubeconfig` directly into the HCP, then that would collide with a secret we already have in the HCP namespace that's related to the guest cluster.

Reply:
done. hard-coded the credentials secret name at the HCP namespace to `kubevirt-infra-credentials`, regardless of the original name at the hostedcluster ("`clusters`") namespace.
it is defined at:
https://github.com/openshift/hypershift/blob/d485b010e270561dda22a5a7f21645a319060dfa/api/v1beta1/hostedcluster_types.go#L138